src/PluginNodes/AggregateStatePluginNode.cs

```diff
@@ -0,0 +1,114 @@
+        private ulong _pressureIndex = 0;
+
+        public void AddOptions(Mono.Options.OptionSet optionSet)
+        {
+            optionSet.Add(
+                "ctas|complextypeaggregatestate",
+                $"add complex type (aggregate state) to address space.\nDefault: {_isEnabled}",
+                (string s) => _isEnabled = s != null);
+        }
+
+        public void AddToAddressSpace(FolderState telemetryFolder, FolderState methodsFolder, PlcNodeManager plcNodeManager)
+        {
+            _plcNodeManager = plcNodeManager;
+
+            if (_isEnabled)
+            {
+                AddNodes(methodsFolder);
+            }
+        }
+
+        public void StartSimulation()
+        {
+            if (_isEnabled)
+            {
+                _nodeGenerator = TimeService.NewTimer(UpdateAggregateState, 5000);
+            }
+        }
+
+        public void StopSimulation()
+        {
+            if (_nodeGenerator != null)
+            {
+                _nodeGenerator.Enabled = false;
+            }
+        }
+
+        private void AddNodes(FolderState methodsFolder)
+        {
+            // Load complex types from binary uanodes file.
+            _plcNodeManager.LoadPredefinedNodes(LoadPredefinedNodes);
+
+            var passiveNode = (BaseObjectState)_plcNodeManager.FindPredefinedNode(
+                new NodeId(AggregateStateModel.Objects.AggregateState1, 
+                _plcNodeManager.NamespaceIndexes[(int)NamespaceType.AggregateState]), 
+                typeof(BaseObjectState));
+
+            // Convert to node that can be manipulated within the server.
+            _node = new AggregateStateState(null);
+            _node.Create(_plcNodeManager.SystemContext, passiveNode);
+
+            _plcNodeManager.AddPredefinedNode(_node);
+
+            Nodes = new List<NodeWithIntervals>
+            {
+                new NodeWithIntervals
+                {
+                    NodeId = "AggregateState",
+                    Namespace = OpcPlc.Namespaces.OpcPlcAggregateState,
+                },
+            };
+        }
+
+        /// <summary>
+        /// Loads a node set from a file or resource and adds them to the set of predefined nodes.
+        /// </summary>
+        private NodeStateCollection LoadPredefinedNodes(ISystemContext context)
+        {
+            var predefinedNodes = new NodeStateCollection();
+
+            predefinedNodes.LoadFromBinaryResource(context,
+                "AggregateState/AggregateStateModel.PredefinedNodes.uanodes", // CopyToOutputDirectory -> PreserveNewest.
+                typeof(PlcNodeManager).GetTypeInfo().Assembly,
+                updateTables: true);
+
+            return predefinedNodes;
+        }
+
+        public void UpdateAggregateState(object state, ElapsedEventArgs elapsedEventArgs)
+        {
+            _pressureIndex++;
+
+            var newValue = new AggregateStateDataType
+            {
+                Temperature = 0,
```

Comment:
The temperature never changes?

Reply:
To change the aggregate state it is possible to define:

* temperature fix and changing only pressure 
* pressure fix and changing only temperatiure
* changing temperature and pressure

For simplicity I used fixed temperature. the other variants can be tested by using OPA UA Clients.

Reply:
Ok, I got confused bcs the description in the readme says "It changes every 5 seconds between different values for temperature and pressure", we might need to clarify that better.

Reply:
Also, pls update the readme - specifically the output of the help under command-line reference. You just need to add the new commands.